src/main/java/edu/ksu/canvas/attendance/repository/AttendanceRepositoryImpl.java

```diff
@@ -40,6 +41,51 @@
         return query.getResultList();
     }
 
+    /**
+     * Gets from the data base all the comments, grouped by student, of one section
+     * @param sectionId
+     * @return a map with pairing a student with all his/her comments
+     */
+    @Override
+    public Map<Long, String> getAttendanceCommentsBySectionId(long sectionId) {
+        Validate.notNull(sectionId, "The sectionId parameter must be not null");
+
+        String sql ="SELECT att.student_id, att.date_of_class, att.notes " +
+                    "FROM attendance att " +
+                    "INNER JOIN attendance_student ats " +
+                    "ON att.student_id = ats.student_id AND ats.canvas_section_id = :canvas_section_id " +
+                    "WHERE att.notes IS NOT NULL ";
+
+        Query query = entityManager.createNativeQuery(sql);
+        query.setParameter("canvas_section_id", sectionId);
+
+        List<Object[]> results = query.getResultList();
+        Map<Long, String> studentCommentsMap = groupCommentsByStudents(results);
```

Comment:
Generally grouping should be done in the sql (or if you decide to do the above change, jpql). This is because databases are made for manipulating data like this, does it very efficiently, and it limits the memory used by the application. And since sql is a declarative language it actually limits the number of possible bugs when manipulating data. (Some states that can happen in Java code just do not exist in SQL. For example, NPEs)

Many of our applications do not do this and instead use daos to load a bunch of data from the database and manipulate it. It is actually probably causing some performance problems currently. 

I wouldn't advise rewriting a bunch of code to do the recommended approach but since you are using sql/jpql in this method anyway it is reasonable to put the grouping there.